backend/src/authentification.ts

```diff
@@ -1,17 +1,39 @@
 import * as express from "express";
 import * as jwt from "jsonwebtoken";
 
+const bannedIP: any = {};
+const toBeBannedIP: any = {};
+
 export const expressAuthentication = (
   request: express.Request,
   securityName: string,
   scopes?: string[]
 ): Promise<any> => {
-  if (securityName === "jwt") {
-    const authHeader = request.headers.Authorization || request.headers.authorization;
 
+  if (securityName === "jwt" || securityName === "tmp") {
+    const authHeader = request.headers.Authorization || request.headers.authorization;
+    const ip = request.ip;
     return new Promise((resolve, reject) => {
       if (!authHeader) {
-        reject(new jwt.JsonWebTokenError("No token provided"));
+        if (securityName === "tmp") {
+          if (bannedIP[ip]) {
+            reject(new jwt.JsonWebTokenError("No token provided and temporary anonymous usage expired, please register with email or wait 4h"));
+          } else {
+            if (!toBeBannedIP[ip]) {
+              toBeBannedIP[ip] = true;
+              setTimeout(() => {
+                bannedIP[ip] = true;
+                setTimeout(() => {
+                  toBeBannedIP[ip] = false;
+                  bannedIP[ip] = false;
+                }, Number(process.env.BACKEND_TMP_WINDOW || "14400") * 1000);
+              }, Number(process.env.BACKEND_TMP_DURATION || "300") * 1000);
```

Comment:
it may be useful to know if the BACKEND_TMP_DURATION is used or if it takes 300 seconds.